UW 537 & UW 538: Add docs for MPAS and MPAS Init drivers. (#450)
ufs-community/uwtools: docs/sections/user_guide/cli/drivers/mpas.rst

``mpas``
==========

> The ``uw`` mode for configuring and running the MPAS forecast model. Any of the ``TASK`` arguments listed below may be called to generate that component required for the ultimate ``run`` task. A ``provisioned_run_directory`` will perform all of the tasks except for running the executable.

Reviewer: ```suggestion
The ``uw`` mode for configuring and running the MPAS forecast model. Each listed ``TASK`` may be called to generate the runtime asset(s) it is responsible for, and will call any task it depends on as needed. A ``provisioned_run_directory`` comprises everything needed for a run, and a ``run`` runs the MPAS executable.
```
I don't know how much better this is. Use any/all/nothing as you like.

Author: This makes a lot more sense! Thanks.

> The ``uw`` mode for configuring and running the MPAS forecast model. Any of the ``TASK`` arguments listed below may be called to generate that component required for a the ultimate ``run`` task. A ``provisioned_run_directory`` will perform all of the tasks except for running the executable.

Author: I don't think we ever really mentioned this anywhere...

I would love feedback on how I could say this more clearly.

> The ``uw`` mode for configuring and running the MPAS forecast model. Any of the ``TASK`` arguments listed below may be called to generate that component required for a the ultimate ``run`` task. A ``provisioned_run_directory`` will perform all of the tasks except for running the executable.

Reviewer 2: "for <s> a</s> the ultimate run"

       namelist_file
         The namelist file
       provisioned_run_directory
         Run directory provisioned with all required content
       run
         A run
       runscript
         The runscript
       streams_file
         The streams file

All tasks take the same arguments. For example:

.. code-block:: text

   $ uw mpas run --help
   usage: uw mpas run --config-file PATH --cycle CYCLE [-h] [--version] [--batch] [--dry-run]
                   [--graph-file PATH] [--quiet] [--verbose]

   A run

   Required arguments:
     --config-file PATH, -c PATH
         Path to config file
     --cycle CYCLE
         The cycle in ISO8601 format

   Optional arguments:
     -h, --help
         Show help and exit
     --version
         Show version info and exit
     --batch
         Submit run to batch scheduler
     --dry-run
         Only log info, making no changes
     --graph-file PATH
         Path to Graphviz DOT output [experimental]
     --quiet, -q
         Print no logging messages
     --verbose, -v
         Print all logging messages

Examples
^^^^^^^^

The examples use a configuration file named ``config.yaml`` with content similar to:

.. highlight:: yaml

> .. literalinclude:: ../../../../shared/mpas.yaml

Author: Okay, so I made the change I suggested above, but instead of using YAML syntax highlighting in the snippet in the text, I just used "text" so that the highlighting would be more consistent between the rendered and templated versions.

Its contents are described in depth in section :ref:`mpas_yaml`.

> * Run ``mpas`` on an interactive node

Reviewer: We write `mpas` in several places here, but I'm not sure it's clear what that refers to. It's the name of our driver, but don't we mean the actual MPAS executable? If so, that's called `atmosphere_model` in the config, but should we maybe just write "MPAS" here and be somewhat generic?

Author: I think it makes sense to refer to MPAS when we want to refer to the executable or the model in general. I will try to go back in and make sure that I use driver in areas where that's what I'm referring to.

> * Run ``mpas`` on an interactive node

Reviewer: ```suggestion
* Run MPAS on an interactive node
```

  .. code-block:: text

     $ uw mpas run --config-file config.yaml --cycle 2025-02-12T12

>   The driver creates a ``runscript.mpas`` file in the directory specified by ``run_dir:`` in the config and runs it, executing ``mpas``.

Reviewer: ```suggestion
  The driver creates a ``runscript.mpas`` file in the directory specified by ``run_dir:`` in the config and runs it, executing ``atmosphere_model``.
```
Assuming we're trying to say the name of the executable that it executes -- which the config shows as `atmosphere_model`.
